rust/src/encodings/plain.rs

```diff
@@ -296,13 +308,16 @@ impl<'a> PlainDecoder<'a> {
         let arrays = stream::iter(chunk_ranges)
             .map(|cr| async move {
                 let index_chunk = indices.slice(cr.start as usize, cr.len());
+                // request contains the array indices we are retrieving in this chunk.
                 let request: &UInt32Array = as_primitive_array(&index_chunk);
 
+                // Get the starting index
                 let start = request.value(0);
+                // Final index is the last value
                 let end = request.value(request.len() - 1);
                 let array = self.get(start as usize..end as usize + 1).await?;
-                let array_byte_boundray = start / 8 * 8_u32;
-                let shifted_indices = subtract_scalar(request, array_byte_boundray)?;
+
+                let shifted_indices = subtract_scalar(request, start)?;
```

Comment:
This change was necessary because Arrow added bounds checking on `take` for boolean arrays.